core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java

```diff
@@ -177,11 +178,15 @@ private Claims parseState(String state)
         }
     }
 
-    public Jws<Claims> parseClaimsJws(String token)
+    public Map<String, Object> convertTokenToClaims(String token)
     {
         return Jwts.parser()
                 .setSigningKeyResolver(signingKeyResolver)
-                .parseClaimsJws(token);
+                .parseClaimsJws(token)
+                .getBody()
```

Comment:
you can just return body because `Claims` implements `Map<String, Object>` if you have further concerns about the exact implementation of the `Claims` object, you could wrapper that chain with `new HashMap<>()`.  Alternatively, you could use `ImmutableMap.copyOf`, but that does not allow null values, which might be a problem here.

Reply:
👍